### PR TITLE
Implement Docs tab end-to-end

### PR DESCRIPTION
**Pull request by cristobalchao**
Includes:

- Implement Categories end-to-end
- Implement URL upload and fetch capabilities
- Integrates fully with BE
 
Screen recording: https://streamable.com/kb5952

### CODE REVIEW

File: wondrous-app/components/AddDocumentDialog/AddDocumentDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+      handleClose();
+      return;
+    }
+
+    createOrgDocument({
+      variables: {
+        input: {
+          ...data,
+          orgId: orgId,
+          categoryId: category.id,
```
Comment (andywong418) [inline wondrous-app/components/AddDocumentDialog/AddDocumentDialog.tsx:83]: Will there always be a category?

File: wondrous-app/components/AddDocumentDialog/AddDocumentDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+          <StyledTextField
+            name="link"
+            placeholder="Enter here a valid link"
+            {...register('link', {
+              required: 'link is required',
+              pattern: { value: URL_REGEX, message: 'Please enter a valid link' },
+            })}
+            fullWidth
+            helperText={errors.link && errors.link.message}
+            error={errors.link}
```
Comment (andywong418) [inline wondrous-app/components/AddDocumentDialog/AddDocumentDialog.tsx:117]: Would we need to guard for the link field if errors are empty?

**Review by andywong418 — COMMENTED** [review]
Looks great - just some questions. Also think there are empty files so we can just remove them to fix lint

File: wondrous-app/components/AddDocumentDialog/AddDocumentDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+
+  const [getOrgDocs] = useLazyQuery(GET_ORG_DOCS, {
+    variables: {
+      orgId,
+    },
+  });
+
+  const [createOrgDocument] = useMutation(CREATE_ORG_DOCUMENT, {
+    onCompleted: () => {
+      getOrgDocs();
```
Comment (andywong418) [inline wondrous-app/components/AddDocumentDialog/AddDocumentDialog.tsx:49]: we could just use `refetchQueries` here

File: wondrous-app/components/AddDocumentDialog/AddDocumentDialog.tsx (hunk 2 above)
Comment (cristobalchao) [reply wondrous-app/components/AddDocumentDialog/AddDocumentDialog.tsx:106]: Good point, added guards

File: wondrous-app/components/AddDocumentDialog/AddDocumentDialog.tsx (hunk 1 above)
Comment (cristobalchao) [reply wondrous-app/components/AddDocumentDialog/AddDocumentDialog.tsx]: Added a conditional clause

**Pull request merged**